### ISSUE TITLE
Multimappers in junction sql

### ISSUE THREAD

**Issue by CharlotteAnne** [langmead-lab/monorail-external#33]
Hi

Thank you for the detailed documentation on multimappers. From what I read it sounds like with Recount3 gene and exon counts will not double count multimapped reads, instead they are semi-randomly allocated to one location as only primary alignments are counted. Am I correct in assuming though that the splice junction files will contain reads counted multiple times if they multimap to multiple junction locations? The STAR documentation states that the sj.out.tab will contain all reads in the output BAM, so I assume this means secondary alignments are also counted in that.

Thanks for any clarification!

Best,
Charlotte

**Comment by ChristopherWilks** [comment]
Hi @CharlotteAnne,

"From what I read it sounds like with Recount3 gene and exon counts will not double count multimapped reads, instead they are semi-randomly allocated to one location as only primary alignments are counted." 

Correct.

"Am I correct in assuming though that the splice junction files will contain reads counted multiple times if they multimap to multiple junction locations? The STAR documentation states that the sj.out.tab will contain all reads in the output BAM, so I assume this means secondary alignments are also counted in that."

Correct, to clarify: multimapping reads *are* included in the `SJ.out.tab` files in recount3/Monorail because we ran STAR with the default `ALL` for `--outSJfilterReads`, https://gensoft.pasteur.fr/docs/STAR/2.7.3a/STARmanual.pdf#subsection.14.16.  However, they're counted separately from uniquely mapping reads by STAR.  So in Monorail/recount3/Snaptron we then summed them together to get the total coverage read count for splice junctions.

I included some of this in my updated explanation (updated just now):
 
https://github.com/langmead-lab/monorail-external/blob/master/MULTIMAPPERS.md#splice-junctions-in-recount3
